Random page API

friendofdog opened the pull request:
- Get a random page from `/page/random/{format}` endpoint
- Add new result types based on return types from Wikipedia API

closes #4

dopecodez commented on source/resultTypes.ts:
```diff
@@ -162,4 +159,133 @@ export interface eventResult {
+  pages: Array<relatedItem>
+}
+
+export interface relatedItem {
```
I think we can use the wikiSummary interface here instead of the relatedItem. Both look almost the same and any fields you feel is missing here should be added to the wikiSummary interface.

friendofdog replied:
Fair point. I removed relatedItem and added the missing fields to wikiSummary.

dopecodez commented:
Other than the changes, the types and the tests look good. Don't worry about the failing coverage request, its a result as per the discussion in #12 , and we will take it up separately.

Once you've tested and feel like this can be merged to master, please go ahead and change the draft status.

You might have to merge the master branch though, I think it has moved ahead of your current branch.

friendofdog commented:
> Other than the changes, the types and the tests look good. Don't worry about the failing coverage request, its a result as per the discussion in #12 , and we will take it up separately.
> 
> Once you've tested and feel like this can be merged to master, please go ahead and change the draft status.
> 
> You might have to merge the master branch though, I think it has moved ahead of your current branch.

Got it.

Added a few tests for different return formats. Not sure how necessary this was, but I figure that the types I just added should be in tests somewhere.

dopecodez commented:
The PR looks good @friendofdog . Thanks for the time you took to document all the types and the comprehensive testing too.

I went ahead and documented the new types and your new method so that we can release this feature as soon as this is merged.

I made a couple of changes in the flow as well. I added a default value as `summary` for our `related` method [here](https://github.com/dopecodez/Wikipedia/pull/11/files#diff-fcd423a0b6ae0654240ee2c1cb0b023707df57b7d1c7a3834c5bfbef7d901da5R534). I also added an input type [`randomFormats`](https://github.com/dopecodez/Wikipedia/pull/11/files#diff-0d1cd8a363968ba8eb861a9dde6aae86df24ada893aeea025e335562f7b66b33L35)
to our method so TypeScript users can see those types available directly. I also added a test case for the same.

I also used your `related` object in one of the available `related` methods on wiki. So, thanks for that too 😄 

We'll go ahead and merge if you feel like these changes are helpful for the users

friendofdog commented:
> The PR looks good @friendofdog . Thanks for the time you took to document all the types and the comprehensive testing too.
> 
> I went ahead and documented the new types and your new method so that we can release this feature as soon as this is merged.
>
> I made a couple of changes in the flow as well. I added a default value as `summary` for our `related` method [here](https://github.com/dopecodez/Wikipedia/pull/11/files#diff-fcd423a0b6ae0654240ee2c1cb0b023707df57b7d1c7a3834c5bfbef7d901da5R534). I also added an input type [`randomFormats`](https://github.com/dopecodez/Wikipedia/pull/11/files#diff-0d1cd8a363968ba8eb861a9dde6aae86df24ada893aeea025e335562f7b66b33L35)
> to our method so TypeScript users can see those types available directly. I also added a test case for the same.
> 
> I also used your `related` object in one of the available `related` methods on wiki. So, thanks for that too 
> 
> We'll go ahead and merge if you feel like my changes work with your method

Just took a look at your changes. Everything makes sense (thanks for explaining things).

I'm not sure if they were missed or were intentionally left out, but I added `mobileSections` and `html`  wherever they seemed to be missing.

dopecodez commented:
The `html` part is a problem @friendofdog . It won't work because we are parsing the [result](https://github.com/dopecodez/Wikipedia/blob/79a4766036a88919987a3214a232ea16a6e2185d/source/request.ts#L49) and this will cause it to return an error.

The `mobileSections` part will work. Adding that makes sense but I think we need to remove the `html`  part for now as having a random html might not be so useful. I don't feels like having a workaround for random in case of random html makes sense.

dopecodez commented on source/optionTypes.ts:
```diff
@@ -32,4 +32,7 @@ export interface eventOptions {
+    'all' | 'selected' | 'births' | 'deaths' | 'events' | 'holidays'
+
+export type randomFormats =
+    'title' | 'summary' | 'html' | 'related' | 'mobile-sections' | 'mobile-sections-lead'
```
Let's remove `html` for now because this will throw an error.

friendofdog replied:
Fixed in latest commit

dopecodez commented on docs/wiki.md:
```diff
@@ -134,10 +134,10 @@ console.log(suggestion); //Returns 'Batman'
 ### random()
 
-Returns a random wiki page in any of the available formats. Formats can be `summary`, `title`, `related`, `mobile-sections` or `mobile-sections-lead`. Defaults to summary by default.
+Returns a random wiki page in any of the available formats. Formats can be `summary`, `title`, `html`, `related`, `mobile-sections` or `mobile-sections-lead`. Defaults to summary.
```
We need to remove `html` from here as well.

friendofdog replied:
Fixed in latest commit

dopecodez commented:
Thanks for the PR @friendofdog  🥇 . You can expect a release with this feature later today.
